core/config/toml/types.go

```diff
@@ -98,6 +98,10 @@ func (c *Core) ValidateConfig() (err error) {
 		err = multierr.Append(err, configutils.ErrInvalid{Name: "RootDir", Value: true, Msg: fmt.Sprintf("Failed to expand RootDir. Please use an explicit path: %s", verr)})
 	}
 
+	if (*c.OCR.Enabled || *c.OCR2.Enabled) && !*c.P2P.V2.Enabled {
+		err = multierr.Append(err, configutils.ErrInvalid{Name: "OCRP2P", Value: true, Msg: fmt.Sprint("P2P required for OCR or OCR2. Please configure P2P or disable OCR/OCR2.")})
```

Comment:
Oh for `Name` we expect a particular TOML key, so maybe we should use:
```suggestion
		err = multierr.Append(err, configutils.ErrInvalid{Name: "P2P.Enabled", Value: false, Msg: fmt.Sprint("P2P required for OCR or OCR2. Please configure P2P or disable OCR/OCR2.")})
```